bitcoin/script/miniscript.cpp

```diff
@@ -402,5 +402,24 @@ bool ParseScriptNumber(const std::pair<opcodetype, std::vector<unsigned char>>&
     return false;
 }
 
+int FindNextChar(const char*& iter, const char* const& end, const char m)
```

Comment:
Agreed, but if it is okay with you, I'll leave this to a follow-up, because I think it'd be worth changing StartsWtih to just use the span Const function too, which will require refactoring Parse to use subspans everywhere instead of `const char* iter`.